### PR TITLE
Fixes for discover search input field (#478)

### LINKED ISSUE #478: discover search input field remains active
**Title:** 	 
- Tapping search icon in Discover brings up the keyboard, but if I swipe left or right to Contacts or Chats the keyboard remains active. 
- The text input field shown in screenshot remains active even after hiding the keyboard (by pressing back on keyboard).

**Device:** 	 samsung SM-G935F, OS Level 23

**Memory:** 	 2814.0/3562.0 MB

**Storage:** 	 19.656/25.389 GB

**Battery:** 	 56, Unplugged

**App Version:** 	 1.0 (1)

**Duration:** 	 161

**Console Log:**
```
 length
11-21 13:51:32.789 E/SpannableStringBuilder(18057): SPAN_EXCLUSIVE_EXCLUSIVE spans cannot have a zero length
11-21 13:51:32.799 E/SpannableStringBuilder(18057): SPAN_EXCLUSIVE_EXCLUSIVE spans cannot have a zero length
11-21 13:51:32.799 E/SpannableStringBuilder(18057): SPAN_EXCLUSIVE_EXCLUSIVE spans cannot have a zero length
11-21 13:51:33.209 D/ViewRootImpl(18057): MSG_RESIZED: ci=Rect(0, 96 - 0, 1036) vi=Rect(0, 96 - 0, 1036) or=1
11-21 13:51:33.229 I/ReactNativeJS(18057): Handling re-frame event: :set
11-21 13:51:33.229 I/ReactNativeJS(18057): Handling re-frame event: :set
11-21 13:51:33.239 I/ReactNativeJS(18057): Handling re-frame event: :update-sync-state
11-21 13:51:33.769 I/ReactNativeJS(18057): Handling re-frame event: :update-sync-state
11-21 13:51:34.269 I/ReactNativeJS(18057): Handling re-frame event: :update-sync-state
11-21 13:51:34.739 D/ViewRootImpl(18057): ViewPostImeInputStage processPointer 0
11-21 13:51:34.799 I/ReactNativeJS(18057): Handling re-frame event: :update-sync-state
11-21 13:51:34.849 D/ViewRootImpl(18057): ViewPostImeInputStage processPointer 1
11-21 13:51:34.869 I/ReactNativeJS(18057): Handling re-frame event: :navigate-to-tab
11-21 13:51:35.259 I/ReactNativeJS(18057): Handling re-frame event: :set-animation
11-21 13:51:35.299 I/ReactNativeJS(18057): Handling re-frame event: :update-sync-state
11-21 13:51:35.849 I/ReactNativeJS(18057): Handling re-frame event: :update-sync-state
11-21 13:51:36.069 D/ViewRootImpl(18057): ViewPostImeInputStage processPointer 0
11-21 13:51:36.179 D/ViewRootImpl(18057): ViewPostImeInputStage processPointer 1
11-21 13:51:36.209 I/ReactNativeJS(18057): Handling re-frame event: :navigate-to-tab
11-21 13:51:36.359 I/ReactNativeJS(18057): Handling re-frame event: :update-sync-state
11-21 13:51:36.629 I/ReactNativeJS(18057): Handling re-frame event: :set-animation
11-21 13:51:36.879 I/ReactNativeJS(18057): Handling re-frame event: :update-sync-state
11-21 13:51:37.389 I/ReactNativeJS(18057): Handling re-frame event: :update-sync-state
11-21 13:51:37.899 I/ReactNativeJS(18057): Handling re-frame event: :update-sync-state
11-21 13:51:38.199 I/art     (18057): Background sticky concurrent mark sweep GC freed 301741(11MB) AllocSpace objects, 41(820KB) LOS objects, 0 free, 41MB/41MB, paused 2.591ms total 174.917ms
11-21 13:51:38.499 I/INSTABUG - a(18057): invokeGeneral
11-21 13:51:38.499 I/Timeline(18057): Timeline: Activity_launch_request id:im.status.ethereum time:377842465
11-21 13:51:38.529 I/Choreographer(18057): Skipped 34 frames!  The application may be doing too much work on its main thread.
11-21 13:51:38.589 D/SensorManager(18057): unregisterListener ::   
11-21 13:51:38.599 I/ReactNativeJS(18057): Handling re-frame event: :update-sync-state
11-21 13:51:38.599 D/RelationGraph(18057): garbageCollect()
11-21 13:51:38.599 W/ResourcesManager(18057): getTopLevelResources: /data/app/im.status.ethereum-1/base.apk / 1.0 running in im.status.ethereum rsrc of package im.status.ethereum

```

**Current View:** 	 im.status.ethereum.MainActivity

[image attachment](https://s3.amazonaws.com/instabug-pro/states/files/37531242/screenshots/d0901fd2b6815cc15d8122fe394a5237_original.jpg?1479707850)

### LINKED ISSUE #483: "Discovery" should be called "Discover"
**Title:** 	 Discovery should be called Discover

**Device:** 	 samsung SM-G935F, OS Level 23

**Memory:** 	 2228.0/3562.0 MB

**Storage:** 	 15.389/25.44 GB

**Battery:** 	 89, Unplugged

**App Version:** 	 1.0 (1)

**Duration:** 	 2

**Console Log:**
```
eJS( 1871): DEBUG [status-im.protocol.web3.transport:16] - :post-message :from 0x04260f5b47cfdc7d7e89e234d8e959aa4b82a909b769ebf2b15e37f332ca8930809eebd80e871d171539628337a17cd1a65c2f4008df9dbfdd007cc89b4b7611a2, :topics [status-discovery-0x04260f5b47cfdc7d7e89e234d8e959aa4b82a909b769ebf2b15e37f332ca8930809eebd80e871d171539628337a17cd1a65c2f4008df9dbfdd007cc89b4b7611a2], :payload 0x7b3a6d6573736167652d69642022313437393831313831333136392d32653237313266632d343462632d356263632d626131312d353066663563663866383861222c203a72657175697265732d61636b3f2066616c73652c203a74797065203a6f6e6c696e652c203a74696d657374616d7020313437393831313831333138372c203a636f6e74656e74206e696c7d0a, :ttl 120
11-22 18:50:21.705 D/ReactNativeJS( 1871): DEBUG [status-im.protocol.web3.delivery:123] - :attempt-was-made 1479811813169-2e2712fc-44bc-5bcc-ba11-50ff5cf8f88a
11-22 18:50:21.715 W/ReactNativeJS( 1871): WARN [status-im.protocol.web3.delivery:132] - :shh-post-error Error: Invalid JSON RPC response: Failed to connect to localhost/127.0.0.1:8545
11-22 18:50:22.165 D/ViewRootImpl( 1871): [ViewRootImpl] action cancel - 1, eccen:3.6666667
11-22 18:50:22.185 D/ReactNativeJS( 1871): DEBUG [status-im.protocol.web3.transport:16] - :post-message :from 0x04260f5b47cfdc7d7e89e234d8e959aa4b82a909b769ebf2b15e37f332ca8930809eebd80e871d171539628337a17cd1a65c2f4008df9dbfdd007cc89b4b7611a2, :topics [status-discovery-0x04260f5b47cfdc7d7e89e234d8e959aa4b82a909b769ebf2b15e37f332ca8930809eebd80e871d171539628337a17cd1a65c2f4008df9dbfdd007cc89b4b7611a2], :payload 0x7b3a6d6573736167652d69642022313437393831313831333136392d32653237313266632d343462632d356263632d626131312d353066663563663866383861222c203a72657175697265732d61636b3f2066616c73652c203a74797065203a6f6e6c696e652c203a74696d657374616d7020313437393831313831333138372c203a636f6e74656e74206e696c7d0a, :ttl 120
11-22 18:50:22.185 D/ReactNativeJS( 1871): DEBUG [status-im.protocol.web3.delivery:123] - :attempt-was-made 1479811813169-2e2712fc-44bc-5bcc-ba11-50ff5cf8f88a
11-22 18:50:22.195 D/ViewRootImpl( 1871): [ViewRootImpl] action cancel - 1, eccen:1.0
11-22 18:50:22.195 D/ViewRootImpl( 1871): ViewPostImeInputStage processPointer 1
11-22 18:50:22.195 D/ViewRootImpl( 1871): ViewPostImeInputStage processPointer 0
11-22 18:50:22.215 D/ViewRootImpl( 1871): ViewPostImeInputStage processPointer 1
11-22 18:50:22.295 W/ReactNativeJS( 1871): WARN [status-im.protocol.web3.delivery:132] - :shh-post-error Error: Invalid JSON RPC response: Failed to connect to localhost/127.0.0.1:8545
11-22 18:50:22.725 I/INSTABUG - a( 1871): invokeGeneral
11-22 18:50:22.735 I/Timeline( 1871): Timeline: Activity_launch_request id:im.status.ethereum time:254712991
11-22 18:50:22.775 D/SensorManager( 1871): unregisterListener ::   
11-22 18:50:22.775 D/RelationGraph( 1871): garbageCollect()
11-22 18:50:22.775 W/ResourcesManager( 1871): getTopLevelResources: /data/app/im.status.ethereum-1/base.apk / 1.0 running in im.status.ethereum rsrc of package im.status.ethereum

```

**Current View:** 	 im.status.ethereum.MainActivity

[image attachment](https://s3.amazonaws.com/instabug-pro/states/files/37709941/screenshots/5cfc696b0bfe32d593c25944a8bb04c9_original.jpg?1479811854)

### PR DESCRIPTION
**Pull request by alwx**
Fixes #478

### CODE REVIEW

**Comment by jarradh** [comment]
I was thinking more like https://github.com/status-im/status-react/tree/bug/discover-typo for permanently removing 'discovery'

**Comment by alwx** [comment]
@jarradh ok, I've renamed the PR

**Comment by jarradh** [comment]
thanks - could you please review #489

File: src/status_im/discovery/screen.cljs [hunk 1]
```diff
@@ -88,21 +91,24 @@
...
-      [discovery-popular {:contacts        contacts
-                          :current-account current-account}]
-      [discovery-recent {:current-account current-account}]]
-     [view contacts-styles/empty-contact-groups
-      ;; todo change icon
-      [icon :group_big contacts-styles/empty-contacts-icon]
-      [text {:style contacts-styles/empty-contacts-text}
-       (label :t/no-statuses-discovered)]])
-   [bottom-gradient]])
+(defn discovery [_]
```
Comment (rasom) [inline src/status_im/discovery/screen.cljs]: Is there a reason to not use `defview`?

**Comment by rasom** [comment]
```
The text input field shown in screenshot remains active even after hiding the keyboard (by pressing back on keyboard).
```
This still happens

**Comment by alwx** [comment]
Brrr, sorry. Now everything is fine.

**Comment by jarradh** [comment]
this seems fine for me

**Pull request merged**